parser/lexer.go

```diff
@@ -527,6 +530,9 @@ func (s *Scanner) scanString() (tok int, pos Pos, lit string) {
 			ch0 = handleEscape(s)
 		}
 		mb.writeRune(ch0, s.r.w)
+		if s.r.eof() {
```

Comment:
use `if s.r.eof() ` is weird, because if check eof() here means that `for !s.r.eof()` have no chance to break this loop..maybe below is better?

```
if !s.r.eof() {
  s.r.inc()
  ch0 = s.r.peek()
}
```